### PR TITLE
chore: Pages context settings

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

Many pages use url-based settings and custom UI controls to alter the component/page properties for manual or automated testing (both integ- and screenshot). While the app-context already simplifies the setup, it still takes too much time to introduce custom controls especially when making quick experiments.

The context settings feature allows enhancing or replacing custom url parameters with a JSON object allowing for quicker experiments setup.

For integration tests the settings JSON property can be passed as url param to set page configuration.

The readonly mode can be used for screenshot testing for the configuration to be visible.

https://github.com/cloudscape-design/components/assets/20790937/07eba5ab-9032-4241-a19f-b7f473116b85

Compared to in-code properties adjustment the playground allows changing properties while preserving the page/component internal state:

https://github.com/cloudscape-design/components/assets/20790937/4cea0403-93aa-4dfc-9987-e0c86801a4b4

### How has this been tested?

Manual testing.

The VR tests fail due to the new header toggle: "Context settings".

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by timogasda — COMMENTED** [review]
Code looks fine, and it's a fun idea :)

Should we have a chat with the rest of team before we're making this "a thing"?

**Comment by pan-kot** [comment]
> Details

Sure, let's discuss!

**Review by just-boris — COMMENTED** [review]
The current implementation only works for basic components without JSX. I think, you cannot use this playground outside button dropdown component, actually

Also, it is unclear what task you solve here. There are two existing approaches how to play with our components:

1. Run locally and make changes right in the pages code
2. Use codesandbox

Why do we need this one?

**Comment by pan-kot** [comment]
> The current implementation only works for basic components without JSX. I think, you cannot use this playground outside button dropdown component, actually
> 
> Also, it is unclear what task you solve here. There are two existing approaches how to play with our components:
> 
> 1. Run locally and make changes right in the pages code
> 2. Use codesandbox
> 
> Why do we need this one?

The primary reason is to simplify creation of pages with settings that alter the component presentation so instead of including checkboxes, dropdowns etc. a JSON can be accepted instead. That is useful for experiments but can also be helpful for test pages that can use different settings coming from the url params.

The settings object does not have to be component props but any custom page configuration. Existing pages like table sticky-columns or autosuggest async test can benefit from having it as well.

File: pages/dropdown/focus-trap.page.tsx [hunk 1]
```diff
@@ -39,44 +27,44 @@ export default function DropdownScenario() {
           <label>
```
Comment (pan-kot) [inline pages/dropdown/focus-trap.page.tsx:27]: The in-page controls can be removed provided the context settings are open during the screenshot tests for better dev experience.

File: src/internal/components/dropdown/__integ__/dropdown-focus-trap.test.ts [hunk 2]
```diff
@@ -13,7 +12,8 @@ interface PageSettings {
 
 function setupTest(settings: PageSettings, testFn: (page: DropdownPageObject) => Promise<void>) {
   return useBrowser(async browser => {
-    await browser.url(`#/light/dropdown/focus-trap?${qs.stringify(settings)}`);
+    await browser.url(`#/light/dropdown/focus-trap?settings=${JSON.stringify({ ...settings, loopFocus: true })}`);
```
Comment (pan-kot) [inline src/internal/components/dropdown/__integ__/dropdown-focus-trap.test.ts:15]: The settings object now needs to be complete, not partial. Unless the defaults handled differently in the page.

**Pull request closed without merging**